src/main/resources/FormItemProfile.json

```diff
@@ -53,6 +59,7 @@
       {
         "fieldName": "Image URI",
         "metadataPropertyName": "image_uri",
+        "description": "The URL that resolves to the image the user of the form is required to annotate in response to the question.  May be a dynamically assigned at run time",
```

Comment:
My understanding was that this profile should be concerned only with the definition of the form, rather than the runtime details, so I am not sure whether we should mention the possibility of a runtime assignment here?

Reply:
Change the field name to 'static image uri'

What do we do about the runtime 'annotate this radiology scan of the patient'

Reply:
?

Reply:
> What do we do about the runtime 'annotate this radiology scan of the patient'

My feeling is that it isn't possible to represent that sort of runtime functionality in the profile, even if it were technically allowed I don't think we'd have a standard way of representing a patient ID in a dynamic URL? Perhaps the best we could do is add a constraint that relates the image URI field to a read-only patient ID field, although I suppose that does imply the dynamic assignment!

cc @jamesrwelch do you possibly have any comment on this, or whether it would be acceptable to have constraints that apply to this URI field (as I'm thinking it's part of the form definition rather than the response)?

Reply:
(this has bearing on drafting of the standard)

Reply:
I think it's the 'reference it somehow' bit that's interesting here.  If the expression that says "give me the patient ID from the response of question 1" is implemented differently in two different forms engines, then we either need to:
- Use our general-purpose constraint language to write this rule, and have the plugin that generates a platform-specific implementation of that.
- or, move the field into a platform-specific profile.
Given that this isn't a use-case we have to deal with a lot at the moment, I'd be ok going with option 1 but leaving it un-implemented for the time-being

Reply:
I think my comment a couple up says we kind of need both - it could be at the NHSE level (find the image in the PCR image cloud), at the app level (say if it was a deployment in CRIS where the API is a known quantity and there is no base URL offset) or even at an _instance_ level - OUH's Millennium talking to CRIS - that some of these parameters are set.

Reply:
Do we have any way to instruct adopters of a logical model what they need to do to successfully turn that model into an implementation model?

Reply:
High level conceptual terms would be fine!!

Maybe its just a standard profile for logical models

Reply:
> Change the field name to 'static image uri'
> 
> What do we do about the runtime 'annotate this radiology scan of the patient'

Thinking about James's comment above, probably the field name is OK as it is, as if we are happy to include the constraint on the image URL then it isn't necessarily static. It is an interesting question though, as the dynamic assignment to the profile value itself is more like an action than a constraint. Perhaps we could still represent both in the same language in the Mauro Rules functionality though.

Reply:
> Is it worthwhile devising a set of attributes that talk about the conceptual kind of data expected here? Or is it just in the notes on the question

By "conceptual data", does this mean properties of the URL such as 'this is dynamic'? I agree that the missing bit for being able to use the standard profile only is the 'reference the value of' functionality/language in the URL string.

Reply:
You might do something as simple as 

{insert-the-url-to-your-pacs-here}/Form("patient_id")

With anything in curly braces turning up in a todo list

Reply:
(where there is a model element called patient_id)

Reply:
I'm happy to park the dynamic URLs problem for the moment.  Make the profile attribute specify its static and we'll add something more for the dynamic problem later

Reply:
That is an excellent example and I'm sure to nick it in the next version of the standard where we discuss this in Annex 2. :-)

I think here you need to put all the semantics you can in the model, and then if you have to dumb down an implementation model then you justify that in your derivation

Reply:
That's also a (possibly unintentional) segue back to the MRC DSS.  You basically riffed on the 'Does this bother you' example on the 'what do you think about the father of your child' form.